File: Problems/783.search-in-a-binary-search-tree.py
# title: Search in a Binary Search Tree
# timestamp: 2025-06-18 00:50:54
# problemUrl: https://leetcode.com/problems/search-in-a-binary-search-tree/
# memory: 19 MB
# runtime: 0 ms

# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def searchBST(self, root: Optional[TreeNode], val: int) -> Optional[TreeNode]:
        while root and root.val != val:
            if root.val < val:
                root = root.right
            else:
                root = root.left
        
        return root

File: Problems/783.search-in-a-binary-search-tree.go
// title: Search in a Binary Search Tree
// timestamp: 2024-04-18 00:47:26
// problemUrl: https://leetcode.com/problems/search-in-a-binary-search-tree/
// memory: 6.8 MB
// runtime: 16 ms

/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func searchBST(root *TreeNode, val int) *TreeNode {
    switch {
        case root == nil:
            return nil
        case root.Val == val:
            return root
        case root.Val > val:
            return searchBST(root.Left, val)
        case root.Val < val:
            return searchBST(root.Right, val)
    }

    return nil
}